crates/frontend/src/components/table.rs

```diff
@@ -7,6 +7,47 @@ use leptos::*;
 use serde_json::{json, Map, Value};
 use superposition_types::SortBy;
 
+#[component]
+pub fn expandable_text(
+    #[prop(into)] text: String,
+    #[prop(into, default = String::new())] cell_class: String,
+    #[prop(default = 100)] max_length: usize,
+) -> impl IntoView {
+    let (is_expanded, set_is_expanded) = create_signal(false);
+
+    view! {
+        <td class=cell_class>
+            <span>
+                {move || {
+                    let should_expand = text.len() > max_length && !is_expanded.get();
+                    let displayed_text = if should_expand {
+                        text[..max_length].to_string()
+                    } else {
+                        text.clone()
+                    };
+                    view! {
+                        <>
+                            {displayed_text}
+                            {if text.len() > max_length {
+                                view! {
+                                    <span
+                                        class="text-blue-500 cursor-pointer ml-2"
+                                        on:click=move |_| set_is_expanded.set(!is_expanded.get())
+                                    >
+                                        {if is_expanded.get() { "Read less" } else { "Read more" }}
```

Comment:
Should we simplify and just show "more" and "less"?
```suggestion
                                        {if is_expanded.get() { "less" } else { "more" }}
```